test/e2e/streaming_test.go

```diff
@@ -275,8 +279,21 @@ func getTLSVolumes(kafkaUserName string) []corev1.Volume {
 }
 
 func waitForKafkaInstance() {
-	err := WaitForStatefulset(t, fw.KubeClient, kafkaNamespace, "my-cluster-kafka", retryInterval, timeout)
-	require.NoError(t, err, "Error waiting for my-cluster-kafka")
+	kafkaInstance := &kafkav1beta1.Kafka{}
+
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		err = fw.Client.Get(context.Background(), types.NamespacedName{Name: "my-cluster", Namespace: kafkaNamespace}, kafkaInstance)
+		require.NoError(t, err)
+
+		for _, condition := range kafkaInstance.Status.Conditions {
+			if strings.EqualFold(condition.Type, "ready") && strings.EqualFold(condition.Status, "true") {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+	require.NoError(t, err)
```

Comment:
Not sure this is required, as the poll doesn't return an error. Possibly the `require.NoError` after the `fw.Client.Get` should just check for error and return it?

Apart from that - will defer to @jpkrohling on whether he is ok with the change in approach.

Reply:
Our func doesn't, but Poll might still return an error, in case the timeout is exceeded. I think it doesn't hurt in having this assertion here.

Reply:
@objectiser The Get will return an error if it doesn't find the Kafka instance at all, in which case the test will just fail immediately.  And @jpkrohling is correct, the require at the end is needed in the case where we find the Kafka instance, but it doesn't go into a ready state within the timeout period.

It's personal preference, but in general I'd rather use require.NoError and fail in the place where an error occurs instead of returning it to the caller.  For the most part if we returned an error to the test case we'd end up just failing there instead.  Calling require.NoError at the actual point of failure makes for shorter test cases that aren't full of if err {} blocks.

Reply:
Fair enough.